### DISCUSSION TITLE
[BUG] rofi doesn't works with sxhkd

### DISCUSSION THREAD

**Discussion by rhs96** [davatorium/rofi#1331] [General]
## Version

Version 1.6.1

## Configuration

Output of `rofi -help` (in a [gist](https://gist.github.com/), please paste the *full* output)
https://gist.github.com/RafaelKudryashov/f66d3c4c6f2a52cf76b9d1af39e331b4

## Launch Command

into sxhkd

`shift + @scape
rofi -show drun` or

`shift + r
	sh ~/.config/rofi/bin/launcher
`

## Steps to reproduce

- Install  rofi with sudo pacman -S rofi
- Define on sxhkd any bind for run rofi
- Tries execute the bind and nothing happens

## What behaviour you see

Nothing happens.
The bind doesn't works. Running the bind command into terminal, will works.

I tested these commands into terminal and works

`sh ~/.config/rofi/bin/launcher` and
`rofi -show drun`

## What behaviour you expect to see

The rofi window.

**Additional details:**
https://github.com/RafaelKudryashov/dotfiles

**Please do not submit reports related to wayland, see
[here](https://github.com/DaveDavenport/rofi/wiki/Wayland) for more
information.**

**Comment by DaveDavenport** [comment]
Can you provide more information?   Does rofi report an error to stdout/stderr when launched via sxhkd.

(I don't use it myself, but people use rofi with sxhkd).

**Reply by rhs96** [reply]
I created my personal script for post install on Arch and i tried several times to run rofi with sxhkd, but without success.
This is my repo: https://github.com/RafaelKudryashov/dotfiles

**Reply by DaveDavenport** [reply]
Can you please provide the requested information?

**Reply by rhs96** [reply]
How I get the stdout/stderr?

**Reply by DaveDavenport** [reply]
With basic shell redirect, and put  it into a file
```
rofi -show run 2>&1 > /tmp/rofi.log 
```

**Reply by rhs96** [reply]
Generates a empty file and opened the rofi.

**Reply by DaveDavenport** [reply]
If you put that in the `sxhkd ` config it works?  that is odd.

**Reply by rhs96** [reply]
Doesn't works into `sxhkd`. 
Only works on `terminal`. Open rofi but don't generate any log record when i put the redirect.

**Reply by DaveDavenport** [reply]
If it works I expect an empty log being created. if it fails I expect some output in the log file.

If you put the redirect in the sxhkd command does it atleast create an empty log? 
(If it doesn't, then the command is not executed at all)

**Reply by rhs96** [reply]
Yeah, the log file is created when I press the bind.

**Reply by DaveDavenport** [reply]
Very weird, rofi is normally noisy when it cannot launch...

Can you change the command in sxhkd too:

G_MESSAGES_DEBUG=all rofi -show drun 2>&1 > /tmp/rofi.log

and pastebin the output?

**Reply by rhs96** [reply]
The output

https://pastebin.com/xqprauYn

**Reply by DaveDavenport** [reply]
That is very odd, and should  not be possible. 
I either expect to see a 'Failed to set locale' error or 'Setup Locale timestamp.'.

There is not a possible other path, unless something internally in setlocale goes wrong.

can you make a keybinding in sxhkd like:

`env > /tmp/env.log`

and paste that? (check for info you don't want to post and anonimize that).

**Reply by DaveDavenport** [reply]
and paste the output from `env` on a terminal.

**Reply by rhs96** [reply]
I'm using a VM at moment. I think it's not necessary to anonymize.

The output from `env.log`.
https://pastebin.com/RnTCfBZZ

The output from `env` on a terminal.
https://pastebin.com/MCFHcuam

**Reply by DaveDavenport** [reply]
I don't see a reason why it should fail in the binding.  I need to reproduce this if I have any hope to figuring this out.
What distribution do you use? and what window manager do you use?

**Reply by rhs96** [reply]
I'm using Arch with BSPWM.

**Reply by DaveDavenport** [reply]
in an empty arch with bspwm install it works fine for me.

**Reply by DaveDavenport** [reply]
I took the image from here:https://gitlab.archlinux.org/archlinux/arch-boxes/-/jobs/24713/artifacts/browse/output
installed minimal dependencies and it worked.

**Reply by DaveDavenport** [reply]
https://beastie.tech/K2QREdCT#arch-2021-06-09_20.15.12.mkv

**Reply by DaveDavenport** [reply]
https://beastie.tech/iZeJRqdz#arch2-2021-06-09_20.22.15.mkv fonts fixed and your launcher script.

**Reply by rhs96** [reply]
Did you install from my dotfiles?
Using setup.sh

**Reply by DaveDavenport** [reply]
not using setup.sh but I did copy the relevant config files.

**Reply by rhs96** [reply]
I'll try to re-install Arch another time and test it with my `setup.sh`. If I have the problem again, I'll give up.

Thanks for the support, Dave!

**Reply by DaveDavenport** [reply]
No problem.  I hope you can get it resolved, please keep me informed.

**Reply by rhs96** [reply]
Hey! Now it's works fine. New Arch installation.

Thanks for the support!

**Comment by DaveDavenport** [comment]
I installed the sxhkd and it works fine for me (the` shift + r rofi -show drun` worked).

**Comment by DaveDavenport** [comment]
Moved to discussion as I don't think this is directly a rofi bug.
